hw/ip/pwrmgr/data/pwrmgr_testplan.hjson

```diff
@@ -11,22 +11,225 @@
+              `otp_idle`, or `flash_idle` inputs off.
+            - Disable all wakeup enables.
+            - Randomly set `wakeup_info_capture_dis` CSR.
+
+            **Checks**:
+            - The `ctrl_cfg_regwen` CSR reads as 1 on the first attempt.
+            - Checks the output `pwr_rst_req.reset_cause` doesn't change for
+              a bounded amount of time.
+            - Check that the `wakeup_info` CSR flags either `fall_through` or
+              `abort` events when capture is enabled.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: reset
+      desc: '''
+            Test random reset and reset_en.
+
+            The different reset inputs can be disabled via bits in the
+            `reset_en` CSR. Resets can be triggered either in active or
+            low power state.
+
+            **Stimulus**:
+            - Sets `reset_en` randomly.
+            - Randomly choose whether to put the unit in low power mode.
+            - Set `rstreqs_i` inputs randomly.
+
+            **Checks**:
+            - Waits for fast fsm to become active reading `ctrl_cfg_regwen`
+              until it becomes 1.
+            - Checks the `reset_status` CSRs.
+            - Checks `ip_clk_en` output has a low transition.
+            - Checks the output `pwr_rst_req.reset_cause` matches a reset
+              cause.
+            - Checks the output `pwr_rst_req.rstreqs` matches the enabled
+              resets.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: escalation_reset
+      desc: '''
+            Test escalation reset.
+
+            An escalation reset cannot be disabled.
+
+            **Stimulus**:
+            - Set `esc_rst_tx_i` inputs randomly.
+
+            **Checks**:
+            - Waits for fast fsm to become active reading `ctrl_cfg_regwen`
+              until it becomes 1.
+            - Checks the `reset_status` CSRs.
+            - Checks the output `pwr_rst_req.reset_cause` matches a reset
+              cause.
+            - Checks the output `pwr_rst_req.rstreqs` is not set.
+            - Check the output `esc_rst_rx_o` indicates an alert escalation.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: reset_wakeup_race
+      desc: '''
+            Test low power entry and reset request almost coinciding.
+
+            If a low power entry and a reset occur at nearly the same time
+            the system handles them one at a time.
+
+            **Stimulus**:
+            - Trigger reset and low power entry as described for other
+              testpoints.
+            - Make sure to test them coinciding also.
+
+            **Check**:
+            - Similar tests as for the wakeup and reset testpoints, except
+              making sure they happen per the triggering order.
+              '''
+      milestone: V2
       tests: []
     }
   ]
+  covergroups: [
+    {
+      name: wakeup_cg
```

Comment:
Could we cross wakeup and reset req occurring at the same time?

Reply:
This is covered in reset_lowpower_distance_cg. Distance zero is on of the possibilities.

Reply:
SG.